lib/types.nix

```diff
@@ -456,6 +456,12 @@ rec {
     # Augment the given type with an additional type check function.
     addCheck = elemType: check: elemType // { check = x: elemType.check x && check x; };
 
+    # Break documentation recursion in infinitely nested submodules.
+    hideSubTree = elemType: elemType // {
+      getSubOptions = prefix: {};
+      getSubModules = null;
+    };
+
```

Comment:
You can just override the description of the type (`type // { description = "custom descr"; }`) to get rid of the infinite recursion, no function in lib needed.

Reply:
This doesn't work. Recursion occurs in manual generation when it tries to generate options description for `locations._name_.locations._name_` etc.

Reply:
There's potentially a more thorough solution now, in 
 - https://github.com/NixOS/nixpkgs/pull/208227

Look for `recursive =` in `types.nix`.

cc @ncfavier